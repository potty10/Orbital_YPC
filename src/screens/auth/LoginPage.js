import React, { useState } from 'react';
import {
  Text, View, TextInput, ImageBackground, Button, KeyboardAvoidingView, Platform,
} from 'react-native';
import { getAuth, signInWithEmailAndPassword } from 'firebase/auth';
import LoginStyles from '../../styles/LoginStyles';
import InlineTextButton from '../../components/InlineTextButton';
import background from '../../assets/background.jpeg';

export default function Login({ navigation }) {
  const auth = getAuth();
  const [errorMessage, setErrorMessage] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const login = async () => {
    if (email === '' || password === '') {
      setErrorMessage('Please enter an email and password');
    } else {
      try {
        await signInWithEmailAndPassword(auth, email, password);
      } catch (error) {
        setErrorMessage(error.message);
      }
    }
  };

  return (
    <ImageBackground style={LoginStyles.imageContainer} source={background}>
      <KeyboardAvoidingView style={LoginStyles.backgroundCover}>
        <Text style={[LoginStyles.lightText, LoginStyles.header]}>Login</Text>
        <Text style={LoginStyles.errorText}>{errorMessage}</Text>
        <TextInput
          style={[LoginStyles.textInput, LoginStyles.lightTextInput, LoginStyles.lightText]}
          placeholder="Email"
          placeholderTextColor="#BEBEBE"
          value={email}
          onChangeText={setEmail}
        />
        <TextInput
          style={[LoginStyles.textInput, LoginStyles.lightTextInput, LoginStyles.lightText]}
          placeholder="Password"
          placeholderTextColor="#BEBEBE"
          secureTextEntry
          value={password}
          onChangeText={setPassword}
        />
        <View style={[LoginStyles.rowContainer, LoginStyles.topMargin]}>
          <Text style={LoginStyles.lightText}>Don&apos;t have an account? </Text>
          <InlineTextButton text="Sign Up" onPress={() => navigation.navigate('SignUp')} />
        </View>
        <View style={[LoginStyles.rowContainer, LoginStyles.bottomMargin]}>
          <Text style={LoginStyles.lightText}>Forgotten your password? </Text>
          <InlineTextButton text="Reset" onPress={() => navigation.navigate('ResetPassword')} />
        </View>
        <Button title="Login" onPress={login} color="#f7b267" />
      </KeyboardAvoidingView>
    </ImageBackground>
  );
}
