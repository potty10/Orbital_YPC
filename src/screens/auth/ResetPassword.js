import React, { useState } from 'react';
import {
  Text, View, TextInput, ImageBackground, Button, KeyboardAvoidingView, Platform,
} from 'react-native';
import { sendPasswordResetEmail, getAuth } from 'firebase/auth';
import LoginStyles from '../../styles/LoginStyles';
import InlineTextButton from '../../components/InlineTextButton';
import background from '../../assets/background.jpeg';

export default function ResetPassword({ navigation }) {
  const auth = getAuth();
  const [email, setEmail] = useState('');
  const [errorMessage, setErrorMessage] = useState('');

  const resetPassword = () => {
    sendPasswordResetEmail(auth, email)
      .then(() => {
        navigation.popToTop();
      })
      .catch((error) => {
        setErrorMessage(error.message);
      });
  };

  return (
    <ImageBackground style={LoginStyles.imageContainer} source={background}>
      <KeyboardAvoidingView style={LoginStyles.backgroundCover}>
        <Text style={[LoginStyles.lightText, LoginStyles.header]}>Reset Password</Text>
        <Text style={LoginStyles.errorText}>{errorMessage}</Text>
        <TextInput
          style={[LoginStyles.textInput, LoginStyles.lightTextInput, LoginStyles.lightText]}
          placeholder="Email"
          placeholderTextColor="#BEBEBE"
          value={email}
          onChangeText={setEmail}
        />
        <View style={[LoginStyles.rowContainer, LoginStyles.bottomMargin]}>
          <Text style={LoginStyles.lightText}>Don&apos;t have an account? </Text>
          <InlineTextButton text="Sign Up" onPress={() => navigation.navigate('SignUp')} />
        </View>
        <Button title="Reset Password" onPress={resetPassword} color="#f7b267" />
      </KeyboardAvoidingView>
    </ImageBackground>
  );
}
