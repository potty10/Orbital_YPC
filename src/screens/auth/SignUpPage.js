import React, { useState } from 'react';
import {
  Text, View, TextInput, ImageBackground, Button, KeyboardAvoidingView, Platform,
} from 'react-native';
import { createUserWithEmailAndPassword, getAuth } from 'firebase/auth';
import LoginStyles from '../../styles/LoginStyles';
import InlineTextButton from '../../components/InlineTextButton';
import background from '../../assets/background.jpeg';

export default function SignUp({ navigation }) {
  const auth = getAuth();

  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [confirmPassword, setConfirmPassword] = useState('');
  const [errorMessage, setErrorMessage] = useState('');

  const validateAndSet = (value, valueToCompare, setValue) => {
    if (value !== valueToCompare) {
      setErrorMessage('Passwords do not match.');
    } else {
      setErrorMessage('');
    }

    setValue(value);
  };

  const signUp = async () => {
    if (password === confirmPassword) {
      try {
        await createUserWithEmailAndPassword(auth, email, password);
        navigation.navigate('Feed');
      } catch (error) {
        setErrorMessage(error.message);
      }
    }
  };

  return (
    <ImageBackground style={LoginStyles.imageContainer} source={background}>
      <KeyboardAvoidingView style={LoginStyles.backgroundCover}>    
        <Text style={[LoginStyles.lightText, LoginStyles.header]}>Sign Up</Text>
        <Text style={[LoginStyles.errorText]}>{errorMessage}</Text>
        <TextInput
          style={[LoginStyles.textInput, LoginStyles.lightTextInput, LoginStyles.lightText]}
          placeholder="Email"
          placeholderTextColor="#BEBEBE"
          value={email}
          onChangeText={setEmail}
        />
        <TextInput
          style={[LoginStyles.textInput, LoginStyles.lightTextInput, LoginStyles.lightText]}
          placeholder="Password"
          placeholderTextColor="#BEBEBE"
          secureTextEntry
          value={password}
          onChangeText={(value) => validateAndSet(value, confirmPassword, setPassword)}
        />
        <TextInput
          style={[LoginStyles.textInput, LoginStyles.lightTextInput, LoginStyles.lightText]}
          placeholder="Confirm Password"
          placeholderTextColor="#BEBEBE"
          secureTextEntry
          value={confirmPassword}
          onChangeText={(value) => validateAndSet(value, password, setConfirmPassword)}
        />
        <View style={[LoginStyles.rowContainer, LoginStyles.bottomMargin]}>
          <Text style={LoginStyles.lightText}>Already have an account? </Text>
          <InlineTextButton text="Login" onPress={() => navigation.popToTop()} />
        </View>
        <Button title="Sign Up" onPress={signUp} color="#f7b267" />
      </KeyboardAvoidingView>
    </ImageBackground>
  );
}
