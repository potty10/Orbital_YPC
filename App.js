// Navigation
// https://reactnavigation.org/docs/stack-navigator/
// react-native-gesture-handler must be imported at the top
import 'react-native-gesture-handler';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';
import { createDrawerNavigator } from '@react-navigation/drawer';

// React
import React from 'react';
import { LogBox } from 'react-native';

// Theme
import { ThemeProvider, useTheme } from '@rneui/themed';
import mainTheme from './src/styles/MainTheme';

// Redux
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import currentWorkoutReducer from './src/slices/currentWorkoutSlice';
import workoutListReducer from './src/slices/workoutListSlice';
import editedWorkoutReducer from './src/slices/editedWorkoutSlice';

// Screens
import Login from './src/screens/auth/LoginPage';
import SignUp from './src/screens/auth/SignUpPage';
import ManageAccount from './src/screens/ManageAccount';
import ResetPassword from './src/screens/auth/ResetPassword';
import UserFeedPage from './src/screens/UserFeedPage';
import ListWorkoutPage from './src/screens/ListWorkoutPage';
import CreateWorkoutPage from './src/screens/CreateWorkoutPage';
import StartWorkoutPage from './src/screens/StartWorkoutPage';
import SearchWorkoutPage from './src/screens/SearchWorkoutPage';
import WorkoutTimerPage from './src/screens/WorkoutTimerPage';
import WorkoutSummaryPage from './src/screens/WorkoutSummaryPage';

// Firebase
import { useAuthentication } from './firebase';

// Ignore log notification by message
// LogBox.ignoreLogs(['Warning: ...']);

// Ignore all log notifications
// LogBox.ignoreAllLogs();

export const store = configureStore({
  reducer: {
    currentWorkout: currentWorkoutReducer,
    workoutList: workoutListReducer,
    editedWorkout: editedWorkoutReducer,
  },
});

const Stack = createStackNavigator();
const Drawer = createDrawerNavigator();

function HomeDrawer() {
  const { theme } = useTheme();
  return (
    <Drawer.Navigator screenOptions={{headerStyle: { backgroundColor: theme.colors.primary}}}>
      <Drawer.Screen name="Feed" component={UserFeedPage} />
      {/* <Drawer.Screen
      name="Workout List"
      component={ListWorkoutPage}
      options={({ navigation }) => ({
        title: 'Workout Plans',
        headerRight: () => (
          <Pressable onPress={() => navigation.navigate('Create Workout')} style={{ marginRight: 26 }}>
            <Text>Create</Text>
          </Pressable>
        ),
      })} /> */}
      <Drawer.Screen name="Workout List" component={ListWorkoutPage} />
      <Drawer.Screen name="Start Workout" component={StartWorkoutPage} />
      <Drawer.Screen name="Manage Account" component={ManageAccount} />
    </Drawer.Navigator>
  );
}

function MainStack() {
  const { theme } = useTheme();
  return (
    <Stack.Navigator screenOptions={{headerStyle: { backgroundColor: theme.colors.primary}}}>
      <Stack.Screen name="Workout Main" component={HomeDrawer} options={{ headerShown: false }} />
      <Stack.Screen name="Create Workout" component={CreateWorkoutPage} />
      <Stack.Screen name="Search Workout" component={SearchWorkoutPage} />
      <Stack.Screen name="Timer Workout" component={WorkoutTimerPage} />
      <Stack.Screen name="Workout Summary" component={WorkoutSummaryPage} />
    </Stack.Navigator>
  )
}
// There are 2 ways to nest navigators and hide parent header
// https://reactnavigation.org/docs/hiding-tabbar-in-screens/
// https://reactnavigation.org/docs/screen-options-resolution (Setting parent screen options based on child navigator's state)
export default function App() {
  const { user } = useAuthentication();

  return (   
      <Provider store={store}>
        <NavigationContainer>
          {user ? (      
            <ThemeProvider theme={mainTheme}> 
                <MainStack/>
            </ThemeProvider>
          ) : (
            <Stack.Navigator>
              <Stack.Screen name="Login" component={Login} options={{ headerShown: false }} />
              <Stack.Screen name="SignUp" component={SignUp} options={{ headerShown: false }} />
              <Stack.Screen name="ResetPassword" component={ResetPassword} options={{ headerShown: false }} />
            </Stack.Navigator>
          )}
        </NavigationContainer>
      </Provider>
    
  );
}
