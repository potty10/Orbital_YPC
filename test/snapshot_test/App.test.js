import React from 'react';
import renderer from 'react-test-renderer';
import App from '../../App';

// Screens
import LoginPage from '../../src/screens/auth/LoginPage'
import ResetPassword from '../../src/screens/auth/ResetPassword'
import SignUpPage from '../../src/screens/auth/SignUpPage'

// https://www.youtube.com/watch?v=VuNPrFH9H0E
// To update all snapshots, run npm test -- -u
// To udpate specific snapshots, run npm test -- path/to/file -u

it('Initial screen', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Login screen', () => {
    const tree = renderer.create(<LoginPage />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('ResetPassword screen', () => {
    const tree = renderer.create(<ResetPassword />).toJSON();
    expect(tree).toMatchSnapshot();
});

it('SignUp screen', () => {
    const tree = renderer.create(<SignUpPage />).toJSON();
    expect(tree).toMatchSnapshot();
});

// it('Initial screen has 1 child', () => {
//     const tree = renderer.create(<App />).toJSON();
//     expect(tree.children.length).toBe(1);
// });