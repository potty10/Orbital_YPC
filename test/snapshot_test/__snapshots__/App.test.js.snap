// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial screen 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        accessibilityIgnoresInvertColors={true}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "#fff",
                            "flex": 1,
                            "justifyContent": "center",
                          }
                        }
                      >
                        <Image
                          source={1}
                          style={
                            Array [
                              Object {
                                "bottom": 0,
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              },
                              Object {
                                "height": undefined,
                                "width": undefined,
                              },
                              undefined,
                            ]
                          }
                        />
                        <View
                          onLayout={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": "stretch",
                              "backgroundColor": "#000000",
                              "justifyContent": "center",
                              "opacity": 0.7,
                              "padding": 16,
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "color": "white",
                                },
                                Object {
                                  "alignSelf": "center",
                                  "fontSize": 18,
                                },
                              ]
                            }
                          >
                            Login
                          </Text>
                          <Text
                            style={
                              Object {
                                "color": "red",
                              }
                            }
                          >
                            
                          </Text>
                          <TextInput
                            onChangeText={[Function]}
                            placeholder="Email"
                            placeholderTextColor="#BEBEBE"
                            style={
                              Array [
                                Object {
                                  "alignSelf": "stretch",
                                  "borderBottomWidth": 2,
                                  "marginVertical": 8,
                                  "padding": 8,
                                },
                                Object {
                                  "borderBottomColor": "white",
                                },
                                Object {
                                  "color": "white",
                                },
                              ]
                            }
                            value=""
                          />
                          <TextInput
                            onChangeText={[Function]}
                            placeholder="Password"
                            placeholderTextColor="#BEBEBE"
                            secureTextEntry={true}
                            style={
                              Array [
                                Object {
                                  "alignSelf": "stretch",
                                  "borderBottomWidth": 2,
                                  "marginVertical": 8,
                                  "padding": 8,
                                },
                                Object {
                                  "borderBottomColor": "white",
                                },
                                Object {
                                  "color": "white",
                                },
                              ]
                            }
                            value=""
                          />
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "stretch",
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                  "marginVertical": 4,
                                },
                                Object {
                                  "marginTop": 16,
                                },
                              ]
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "white",
                                }
                              }
                            >
                              Don't have an account? 
                            </Text>
                            <View
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#87F1FF",
                                  }
                                }
                              >
                                Sign Up
                              </Text>
                            </View>
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "alignSelf": "stretch",
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                  "marginVertical": 4,
                                },
                                Object {
                                  "marginBottom": 16,
                                },
                              ]
                            }
                          >
                            <Text
                              style={
                                Object {
                                  "color": "white",
                                }
                              }
                            >
                              Forgotten your password? 
                            </Text>
                            <View
                              accessible={true}
                              collapsable={false}
                              focusable={true}
                              onBlur={[Function]}
                              onClick={[Function]}
                              onFocus={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                            >
                              <Text
                                style={
                                  Object {
                                    "color": "#87F1FF",
                                  }
                                }
                              >
                                Reset
                              </Text>
                            </View>
                          </View>
                          <View
                            accessibilityRole="button"
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {},
                                ]
                              }
                            >
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#007AFF",
                                      "fontSize": 18,
                                      "margin": 8,
                                      "textAlign": "center",
                                    },
                                    Object {
                                      "color": "#f7b267",
                                    },
                                  ]
                                }
                              >
                                Login
                              </Text>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;

exports[`Login screen 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#000000",
        "justifyContent": "center",
        "opacity": 0.7,
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
          },
          Object {
            "alignSelf": "center",
            "fontSize": 18,
          },
        ]
      }
    >
      Login
    </Text>
    <Text
      style={
        Object {
          "color": "red",
        }
      }
    >
      
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#BEBEBE"
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "borderBottomWidth": 2,
            "marginVertical": 8,
            "padding": 8,
          },
          Object {
            "borderBottomColor": "white",
          },
          Object {
            "color": "white",
          },
        ]
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#BEBEBE"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "borderBottomWidth": 2,
            "marginVertical": 8,
            "padding": 8,
          },
          Object {
            "borderBottomColor": "white",
          },
          Object {
            "color": "white",
          },
        ]
      }
      value=""
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 4,
          },
          Object {
            "marginTop": 16,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Don't have an account? 
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#87F1FF",
            }
          }
        >
          Sign Up
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 4,
          },
          Object {
            "marginBottom": 16,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Forgotten your password? 
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#87F1FF",
            }
          }
        >
          Reset
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#f7b267",
              },
            ]
          }
        >
          Login
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`ResetPassword screen 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#000000",
        "justifyContent": "center",
        "opacity": 0.7,
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
          },
          Object {
            "alignSelf": "center",
            "fontSize": 18,
          },
        ]
      }
    >
      Reset Password
    </Text>
    <Text
      style={
        Object {
          "color": "red",
        }
      }
    >
      
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#BEBEBE"
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "borderBottomWidth": 2,
            "marginVertical": 8,
            "padding": 8,
          },
          Object {
            "borderBottomColor": "white",
          },
          Object {
            "color": "white",
          },
        ]
      }
      value=""
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 4,
          },
          Object {
            "marginBottom": 16,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Don't have an account? 
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#87F1FF",
            }
          }
        >
          Sign Up
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#f7b267",
              },
            ]
          }
        >
          Reset Password
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`SignUp screen 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#000000",
        "justifyContent": "center",
        "opacity": 0.7,
        "padding": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
          },
          Object {
            "alignSelf": "center",
            "fontSize": 18,
          },
        ]
      }
    >
      Sign Up
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "red",
          },
        ]
      }
    >
      
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#BEBEBE"
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "borderBottomWidth": 2,
            "marginVertical": 8,
            "padding": 8,
          },
          Object {
            "borderBottomColor": "white",
          },
          Object {
            "color": "white",
          },
        ]
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#BEBEBE"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "borderBottomWidth": 2,
            "marginVertical": 8,
            "padding": 8,
          },
          Object {
            "borderBottomColor": "white",
          },
          Object {
            "color": "white",
          },
        ]
      }
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Confirm Password"
      placeholderTextColor="#BEBEBE"
      secureTextEntry={true}
      style={
        Array [
          Object {
            "alignSelf": "stretch",
            "borderBottomWidth": 2,
            "marginVertical": 8,
            "padding": 8,
          },
          Object {
            "borderBottomColor": "white",
          },
          Object {
            "color": "white",
          },
        ]
      }
      value=""
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginVertical": 4,
          },
          Object {
            "marginBottom": 16,
          },
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Already have an account? 
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <Text
          style={
            Object {
              "color": "#87F1FF",
            }
          }
        >
          Login
        </Text>
      </View>
    </View>
    <View
      accessibilityRole="button"
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#f7b267",
              },
            ]
          }
        >
          Sign Up
        </Text>
      </View>
    </View>
  </View>
</View>
`;
